mParticle-Apple-SDK/Listener/MPListenerController.swift

```diff
@@ -0,0 +1,353 @@
+     * than the method containing this method's invocation
+     * @param methodName the name of the Kit's method being called
+     * @param kitId the Id of the kit
+     * @param used whether the Kit's method returned ReportingMessages, or null if return type is void
+     * @param objects the arguments supplied to the Kit
+     */
+    @objc optional func onKitApiCalled(_ methodName: Any!, kitId: Int32, used: Any!, objects: Any!)
+
+    /**
+     * Indicates that a Kit module, with kitId, has been included in the source files
+     * @param kitId the id of the kit, corresponds with a {@see MPKitInstance}
+     */
+    @objc optional func onKitDetected(_ kitId: Int32)
+
+    /**
+     * Indicates that a Configuration for a kit with kitId is being applied
+     * @param kitId the id of the kit, corresponds with a {@see MPKitInstance}
+     * @param configuration the kit
+     */
+    @objc optional func onKitConfigReceived(_ kitId: Int32, configuration: Any!)
+
+    /**
+     * Indicates that a kit with kitId was successfully started
+     * @param kitId the id of the kit, corresponds with a {@see MPKitInstance}
+     */
+    @objc optional func onKitStarted(_ kitId: Int32)
+
+    /**
+     * Indicates that either an attempt to start a kit was unsuccessful, or a started kit was stopped.
+     * Possibilities for why this may happen include: {@see MParticleUser}'s loggedIn status or
+     * {@see MPConsentState} required it to be stopped, the Kit crashed, or a
+     * configuration was received that excluded the kit
+     * @param kitId the id of the kit, corresponds with a {@see MPKitInstance}
+     * @param reason a message containing the reason a kit was stopped
+     */
+    @objc optional func onKitExcluded(_ kitId: Int32, reason: Any!)
+
+    /**
+     * Indicates that state of a Session may have changed
+     * @param session the current {@see MParticleSession} instance
+     */
+    @objc optional func onSessionUpdated(_ session: Any!)
+
+    /**
+     * Indicates that an alias request has completed
+     * @param aliasResponse the alias response object
+     */
+    @objc optional func onAliasRequestFinished(_ aliasResponse: MPAliasResponse?)
+}
+
+
+@objc public class MPListenerController : NSObject {
+    private static let shared = MPListenerController()
+    private var sdkListeners: [MPListenerProtocol] = []
+
+    /**
+     * Returns the shared instance object.
+     * @returns the Singleton instance of the MPListener class.
+     */
+    @objc public class func sharedInstance() -> MPListenerController {
+        return shared
+    }
+
+    /**
+     * Adds a listener to the SDK to receive any MPListenerProtocol calls from the API to that object
+     * @param sdkListener An instance of a class that implements the MPListenerProtocol
+     */
+    @objc public func addSdkListener(_ sdkListener: any MPListenerProtocol) {
+        self.sdkListeners.append(sdkListener)
```

Comment:
```suggestion
        sdkListeners.append(sdkListener)
```

Per style guide, let's get rid of these explicit `self.` calls in